client/StatusTable/saga.js

```diff
@@ -0,0 +1,71 @@
+import {takeLatest, call, put, select, all} from 'redux-saga/effects'
+import * as actions from './actions'
+import {  getWorkdaysForMonth } from '../api/mongoService'
+import { finish } from "../api/timeService";
+import {getStudentId} from "../rootSelectors";
+import WeekRow from "./components/WeekRow";
+import React from "react";
+
+export default  function* statusTableSaga() {
+    yield all([
+        takeLatest(actions.tableCloseInit, finishSaga),
+        takeLatest(actions.tableLoadInit, startSaga)
+    ])
+}
+
+// react on LogOut
+function* finishSaga() {
+    try {
+        const _id = yield select(getStudentId);
+
+        yield put(actions.tableCloseRequest());
+        yield call(finish, _id);
```

Comment:
we decided don't finish working time on logout. Anyway it shouldn't be handled in the table saga it is responsibility of login-logout part